### PR TITLE
feat(KP-373): replace the old search with the newer version

### PR DESCRIPTION
**Pull request by amirhossein-haerian**
In this commit the old search and all of the dependencies are removed now.

### CODE REVIEW

**Review by axelbjo — APPROVED** [review]
Looks good to me! I tried running this on my machine as well and it seems to work fine.

File: public/js/app/pages/SearchLandingPage.tsx
Comment (karlandindrakryggen) [inline public/js/app/pages/SearchLandingPage.tsx:1]: The component is still called `NewSearchLandingPage` internally here.

File: public/js/app/pages/SearchPage.tsx
Comment (karlandindrakryggen) [inline public/js/app/pages/SearchPage.tsx:1]: The component is still called `NewSearchPage` internally here.

File: server/server.js [hunk 1]
```diff
@@ -252,27 +252,20 @@ appRoute.get(
...
-  Search.searchThirdCycleCourses
+  proxyPrefixPath.thirdCycleCourseSearch,
+  SearchPage.searchThirdCycleCourses
 )
 appRoute.get(
   'public.NewSearchThirdCycleCoursesResult',
-  proxyPrefixPath.thirdCycleCourseSearchResultNew,
-  Search.searchThirdCycleCourses
+  proxyPrefixPath.thirdCycleCourseSearchResult,
+  SearchPage.searchThirdCycleCourses
```
Comment (karlandindrakryggen) [inline server/server.js:265]: Some "newSearch..." left here to rename:
- public.newSearchAllCourses
- public.newSearchAllCoursesResult
- public.NewSearchThirdCycleCourses
- public.NewSearchThirdCycleCoursesResult

**Review by karlandindrakryggen — COMMENTED** [review]
Find some files/components/strings with the word "newSearch..." still. 

Both `searchHelper.js` and `newSearchHelper.ts` still exists. Remove the first one, and rename the second to just searchHelper.ts? 

And see more places in comments:

File: server/controllers/searchCtrl.js [hunk 2]
```diff
@@ -1,167 +0,0 @@
...
-
-  const { query } = req
-  // Example: `text_pattern=${pattern}`
-  const searchParamsStr = stringifyKoppsSearchParams(query)
-
-  try {
-    log.debug(` trying to perform a search of courses with ${searchParamsStr} transformed from parameters: `, { query })
-
-    const apiResponse = await koppsApi.getSearchResults(searchParamsStr, lang)
-    log.debug(` performCourseSearch with ${searchParamsStr} response: `, apiResponse)
```
Comment (karlandindrakryggen) [inline server/controllers/searchCtrl.js:155]: `getSearchResults` is only used here, so it could be remove from `koppsApi.js` now.

**Review by karlandindrakryggen — CHANGES_REQUESTED** [review]
Looks great! Some renaming left (see previous comment) and a small possible cleanup in koppsApi.js.

**Comment by amirhossein-haerian** [comment]
> Looks great! Some renaming left (see previous comment) and a small possible cleanup in koppsApi.js.

Thanks for checking the files and for your great comments, now I made the required changes and updated the code.

**Review by karlandindrakryggen — APPROVED** [review]

**Pull request merged**